logsapi/event.go

```diff
@@ -101,6 +108,18 @@ func (lc *Client) ProcessLogs(
 					lc.logger.Warn("report event request id didn't match the previous event id")
 					lc.logger.Debug("Log API runtimeDone event request id didn't match")
 				}
+			case FunctionLog:
+				if metadataContainer != nil && len(metadataContainer.Metadata) > 0 {
```

Comment:
[**For reviewers**]: The way we collect metadata currently allows function logs to be emitted before metadata can be collected. Since APM Server considers any request without metadata to be `400` the PR drops any log lines before metadata is collected. Another approach can be to buffer these log lines and send them in batch but that might increase the resource utilization of the extension. Let me know if there are any concerns/alternative approaches here.